src/renderer/webgl/shaders/RadialGradientShader.mjs

```diff
@@ -63,73 +49,116 @@ export default class RadialGradientShader extends DefaultShader {
+        if(Array.isArray(v) && v.length === 2) {
+            this._pivot = v;
+        }
+        else if(Array.isArray(v)) {
+            this._pivot = [v[0], v[1] || v[0]];
+        }
+        else {
+            this._pivot = [v, v];
         }
+        this.redraw();
     }
 
-    setupUniforms(operation) {
-        super.setupUniforms(operation);
-        // We substract half a pixel to get a better cutoff effect.
-        const rtc = operation.getNormalRenderTextureCoords(this._x, this._y);
-        this._setUniform("center", new Float32Array(rtc), this.gl.uniform2fv);
+    get pivot() {
+        return this._pivot[0];
+    }
 
-        this._setUniform("radius", 2 * this._radiusX / operation.getRenderWidth(), this.gl.uniform1f);
+    set pivotY(f) {
+        this._pivot[1] = f;
+        this.redraw();
+    }
 
+    get pivotY() {
+        return this._pivot[1];
+    }
 
-        // Radial gradient shader is expected to be used on a single element. That element's alpha is used.
-        this._setUniform("alpha", operation.getElementCore(0).renderContext.alpha, this.gl.uniform1f);
+    set pivotX(f) {
+        this._pivot[0] = f;
+        this.redraw();
+    }
 
-        this._setUniform("color", this._rawColor, this.gl.uniform4fv);
-        this._setUniform("aspectRatio", (this._radiusX/this._radiusY) * operation.getRenderHeight()/operation.getRenderWidth(), this.gl.uniform1f);
+    get pivotX() {
+        return this._pivot[0];
     }
 
-}
+    _getNormalizedColor(color) {
+        const col = StageUtils.getRgbaComponentsNormalized(color);
+        col[0] *= col[3];
+        col[1] *= col[3];
+        col[2] *= col[3];
+        return new Float32Array(col);
+    }
 
-RadialGradientShader.vertexShaderSource = `
-    #ifdef GL_ES
-    # ifdef GL_FRAGMENT_PRECISION_HIGH
-    precision highp float;
-    # else
-    precision lowp float;
-    # endif
-    #endif
-    attribute vec2 aVertexPosition;
-    attribute vec2 aTextureCoord;
-    attribute vec4 aColor;
-    uniform vec2 projection;
-    uniform vec2 center;
-    uniform float aspectRatio;
-    varying vec2 pos;
-    varying vec2 vTextureCoord;
-    varying vec4 vColor;
-    void main(void){
-        gl_Position = vec4(aVertexPosition.x * projection.x - 1.0, aVertexPosition.y * -abs(projection.y) + 1.0, 0.0, 1.0);
-        vTextureCoord = aTextureCoord;
-        vColor = aColor;
-        gl_Position.y = -sign(projection.y) * gl_Position.y;
-        pos = gl_Position.xy - center;
-        pos.y = pos.y * aspectRatio;
+    setupUniforms(operation) {
+        super.setupUniforms(operation);
+        const owner = operation.shaderOwner;
+
+        if(this._x) {
+            this._pivot[0] = this._x / owner.w;
+        }
+        if(this._y) {
+            this._pivot[1] = this._y / owner.h;
+        }
+
+        if(this._radius === 0) {
+            this._radius = owner.w * 0.5;
+        }
+
+        this._setUniform('innerColor', this._normalizedIC, this.gl.uniform4fv);
```

Comment:
Please note that on some of the platforms it is required to convert arrays to `Float32Array` for shader to work.